Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -101,22 +102,32 @@ private func createResolvedPackages(allManifests: [Manifest], manifestToPackage:
         // Get all the external dependencies of this package.
         let dependencies = manifest.package.dependencies.map{ packageURLMap[$0.url]! }
 
-        // FIXME: Temporary until we switch to product based dependencies.
-        let externalModuleDependencies = dependencies.flatMap{ $0.modules.filter{ $0.type != .test } }
-
         // Topologically Sort all the local modules in this package.
         let modules = try! topologicalSort(package.modules, successors: { $0.dependencies })
 
         // Make sure these module names are unique in the graph.
-        if let duplicateModules = externalModuleDependencies.lazy.map({$0.name}).duplicates(modules.lazy.map{$0.name}) {
+        let dependencyModuleNames = dependencies.lazy.flatMap{ $0.modules }.flatMap{ $0.name }
+        if let duplicateModules = dependencyModuleNames.duplicates(modules.lazy.map{$0.name}) {
             throw ModuleError.duplicateModule(duplicateModules.first!)
         }
 
+        // Add system module dependencies directly to the target's dependencies because they are 
+        // not representable as a product.
```

Comment:
I wonder if we should treat system module dependencies as some kind of implicit product internally, just to make things more consistent?